rest/handler.go

```diff
@@ -82,27 +83,28 @@ const (
 	regularPrivs = iota // Handler requires valid authentication
 	publicPrivs         // Handler Handler checks auth and falls back to guest if invalid or missing
 	adminPrivs          // Handler ignores auth, always runs with root/admin privs
+	metricsPrivs
 )
 
 type handlerMethod func(*handler) error
 
 // Creates an http.Handler that will run a handler with the given method
-func makeHandler(server *ServerContext, privs handlerPrivs, method handlerMethod) http.Handler {
+func makeHandler(server *ServerContext, privs handlerPrivs, checkPermissions []string, storePermissionsResults bool, method handlerMethod) http.Handler {
```

Comment:
What does `storePermissionResults` do and how do I know whether I want to set it or not for a given route? I can't tell what the `permissionsResults` map is used for...

Reply:
`storePermissionResults` defines whether to store the results of the permissions check on the handler. This will allow handlers to figure out what to do based on any permissions it has. For example, the GET _user will do different things based on different permissions passed in. Although given the cost I might just save results for all of them.

Reply:
As discussed, may want to differentiate between endpoint access permissions, and endpoint response permissions, as it's only the latter we need to store.